src/libraries/System.Runtime/tests/System/DateTimeTests.cs

```diff
@@ -1041,6 +1041,10 @@ public static void Parse_InvalidArguments_Throws()
             Assert.Throws<FormatException>(() => DateTime.Parse(""));
             Assert.Throws<FormatException>(() => DateTime.Parse("", new MyFormatter()));
             Assert.Throws<FormatException>(() => DateTime.Parse("", new MyFormatter(), DateTimeStyles.NoCurrentDateDefault));
+
+            Assert.Throws<FormatException>(() => DateTime.Parse("2020-5-7T09:37:00.0000000-07:00c"));
+            Assert.Throws<FormatException>(() => DateTime.Parse("2020-5-7T09:37:00.0000000-07:00c", new MyFormatter()));
+            Assert.Throws<FormatException>(() => DateTime.Parse("2020-5-7T09:37:00.0000000-07:00c", new MyFormatter(), DateTimeStyles.NoCurrentDateDefault));
```

Comment:
these should continue to work (per the strange treatment of # and \0), so you might want to add a test to prove it was not broken by your change:
`2020-5-7T09:37:00.0000000+00:00\0`
`#2020-5-7T09:37:00.0000000+00:00#`
`#2020-5-7T09:37:00.0000000+00:00#\0`
I only see tests for these without the timezone part.
These should still fail
`2020-5-7T09:37:00.0000000+00:00#`
`2020-5-7T09:37:00.0000000+00:00#\0`

Reply:
I second @danmosemsft for adding more test cases for valid formats. Also, please ensure adding cases when there is nothing at the end too. something like `2020-5-7T09:37:00.0000000-07:00`

Reply:
@danmosemsft @tarekgh Fixed, please review, thanks.